app/process_proposal.go

```diff
@@ -104,7 +104,7 @@ func (app *App) ProcessProposal(req abci.RequestProcessProposal) (resp abci.Resp
 	}
 
 	// Construct the data square from the block's transactions
-	dataSquare, err := square.Construct(req.BlockData.Txs, app.GetBaseApp().AppVersion(), app.GovSquareSizeUpperBound(sdkCtx))
+	dataSquare, err := square.Construct(req.BlockData.Txs, app.GetBaseApp().AppVersion())
```

Comment:
[blocking] How do we enforce the governance based square size if we don't pass it here in Construct? From looking at this change, it seems a proposer could build an invalid block that is larger than the govmaxsquaresize and still get it committed.